zio/bzngio/index.go

```diff
@@ -0,0 +1,175 @@
+package bzngio
+
+import (
+	"errors"
+	"io"
+	"os"
+
+	"github.com/brimsec/zq/pkg/nano"
+	"github.com/brimsec/zq/zbuf"
+	"github.com/brimsec/zq/zng"
+	"github.com/brimsec/zq/zng/resolver"
+)
+
+type Index struct {
```

Comment:
Maybe a comment indicating that this is a time index? Or call the struct TimeIndex? I guess the high-level bit is given the other kinds of indexing we have (or will have) elsewhere, it seems like the time-specific nature of this should be highlighted somehow.